cpp/CMakeLists.txt

```diff
@@ -48,7 +48,9 @@ set(CMAKE_CUDA_STANDARD 14)
 set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+    # Suppress parentheses warning which causes gmock to fail
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-parentheses")
```

Comment:
I don't want us to get in the habit of disabling warnings. Can you file an issue against GTest?

Reply:
Ok, I'll dig into this warning a bit further. The line that was failing didn't actually contain parentheses, so maybe something is wrong on our end

Reply:
It seems to be CUDA related. I'm wondering if nvcc is doing some preprocessing that inserts parentheses. Here's a related issue: https://github.com/ECP-copa/Cajita/issues/2

Reply:
The nvcc preprocessing converts `const FieldType Class::*field_;` into `const FieldType (Class::*field_);`, which `-Werror=parentheses` flags as an error. I was directed to file a bug against CUDA.

I can remove `-Wno-parentheses` from CMAKE_CXX_FLAGS since this only affects CUDA. That said, I understand not wanting to disable warnings, but I think gcc went a little too far with warnings about having "unnecessary" parentheses.